### ISSUE TITLE
I upgrade to WSL2 backend, now have to reset to factory default to get Docker to start each time

### ISSUE THREAD

**Issue by flickerfly** [docker/for-win#6961]
- [ X ] I have tried with the latest version of my channel (Stable or Edge)
  - [ X ] I have uploaded Diagnostics
  - Diagnostics ID: 1BF81BCB-7B29-4E41-9E69-C743C2C277E4/20200529202123

### Expected behavior
Docker starts upon Windows starting back up.

### Actual behavior
Docker Desktop sits for ~10min saying it is starting Docker. Then it throws a crash report as in the uploaded Diagnostics.

```
System.TimeoutException:
Timed out while wait for Docker service to be ready
   at Docker.LinuxkitDaemonStartup.<StartAsync>d__5.MoveNext() in C:\workspaces\stable-2.3.x\src\github.com\docker\pinata\win\src\Docker.Desktop\LinuxkitDaemonStartup.cs:line 43
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Docker.Engines.WSL2.LinuxWSL2Engine.<DoStartAsync>d__23.MoveNext() in C:\workspaces\stable-2.3.x\src\github.com\docker\pinata\win\src\Docker.Desktop\Engines\WSL2\LinuxWSL2Engine.cs:line 149
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Docker.ApiServices.StateMachines.TaskExtensions.<WrapAsyncInCancellationException>d__0.MoveNext() in C:\workspaces\stable-2.3.x\src\github.com\docker\pinata\win\src\Docker.ApiServices\StateMachines\TaskExtensions.cs:line 29
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Docker.ApiServices.StateMachines.StartTransition.<DoRunAsync>d__5.MoveNext() in C:\workspaces\stable-2.3.x\src\github.com\docker\pinata\win\src\Docker.ApiServices\StateMachines\StartTransition.cs:line 67
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at Docker.ApiServices.StateMachines.StartTransition.<DoRunAsync>d__5.MoveNext() in C:\workspaces\stable-2.3.x\src\github.com\docker\pinata\win\src\Docker.ApiServices\StateMachines\StartTransition.cs:line 92
```

### Information

  - Windows Version: Version	10.0.19041 Build 19041
  - Docker Desktop Version: 2.3.0.3 (45519) stable channel
  - Are you running inside a virtualized Windows e.g. on a cloud server or on a mac VM: No

This problem started yesterday with a feature upgrade of Windows that enabled WSL2. When Docker Desktop suggested I enable that, I did. It seems to occur each time I restart. In the logs I did see some notes about port 2375 contention so tried some steps I saw related to that and Hyper-V, but I just left Hyper-V off since I'm trying to move all WSL2 

I've disabled the Microsoft-Hyper-V feature. I tried to excludeportrange with this command, but saw the following failure.
```
PS C:\WINDOWS\system32> Get-WindowsOptionalFeature -online -featurename Microsoft-Hyper-V
FeatureName      : Microsoft-Hyper-V
DisplayName      : Hyper-V Platform
Description      : Provides the services that you can use to create and manage virtual machines and their resources.
RestartRequired  : Possible
State            : Disabled
CustomProperties :

PS C:\WINDOWS\system32> netsh int ipv4 add excludedportrange protocol=tcp startport=2375 numberofports=1
The process cannot access the file because it is being used by another process.
```

As far as that port is concerned.
```
$ netstat -a -o|Select-String -Pattern 2375
$ netsh interface ipv4 show excludedportrange protocol=tcp

Protocol tcp Port Exclusion Ranges

Start Port    End Port
----------    --------
      1303        1402
      1403        1502
      1503        1602
      1603        1702
      1703        1802
      1803        1902
      1903        2002
      2003        2102
      2103        2202
      2203        2302
      2303        2402
      2403        2502
     50000       50059     *

* - Administered port exclusions.
```

### Steps to reproduce the behavior

  1. For me it was upgrade to Windows 2004
  2. Enable WSL2
  3. Restart system

**Comment by flickerfly** [comment]
Some research and troubleshooting has show that unsetting "Expose daemon on TCP://localhost:2375 without TLS" allows Docker Desktop to start, but that removes how I get to it from my WSL instances so looking into alternatives for that. Looks like a lot of other related tickets also. 

Likely related to this:
#6962 
#6905 
#6895 

Harder to say, but maybe related:
#6944 
#6940 
#6931 
#6929 
#6877 

I'm running stable 2.3.0.3 according to the "About Docker Desktop" menu item, but when I tried to upgrade to Edge 2.3.1.3 (saw a suggestion that might be helpful in another issue) it downloads 2.3.1.0 and then says I already have 2.3.1.0 installed. Then I restart Docker Desktop and it still says 2.3.0.3.

**Comment by StefanScherer** [comment]
Are you able to run other WSL 2 distros?
What is `wsl -l -v` showing you?

When you work in WSL 2 distros you normally do not need the port 2375, you can enable the WSL integration in the Docker Desktop settings -> Resources. https://docs.docker.com/docker-for-windows/wsl/ shows you a screenshot of that. And it's hard to use the port 2375 within another WSL 2 distro as WSL 2 uses a different IP address.

**Comment by stephen-turner** [comment]
In the latest Edge release, 2.3.2.0, we have improved this so that application startup is no longer blocked when port 2375 is in use, and a toast notification pops up instead.

**Comment by cybernetics** [comment]
same here. not able to start docker for desktop. using win 10 Pro (v 2004) with WSL2. during the installation I did not select the checkbox to enable WSL2 backend, still, it seems like it's trying to use WSL2 backend.

also submitted docker diagnostic with id: 96DD888D-3B30-4F40-A5F5-BC139A7300EB/20200629051046

getting the following error during startup:

System.InvalidOperationException:
Failed to deploy distro docker-desktop to %localappdata%\Docker\wsl\distro: exit code: -1
 stdout: Error: 0xffffffff


 stderr: 
   at Docker.Core.WslShortLivedCommandResult.LogAndThrowIfUnexpectedExitCode(String prefix, ILogger log, Int32 expectedExitCode) in C:\workspaces\master\src\github.com\docker\pinata\win\src\Docker.Core\WslCommand.cs:line 119
   at Docker.Engines.WSL2.WSL2Provisioning.<DeployDistroAsync>d__16.MoveNext() in C:\workspaces\master\src\github.com\docker\pinata\win\src\Docker.Desktop\Engines\WSL2\WSL2Provisioning.cs:line 142
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Docker.Engines.WSL2.WSL2Provisioning.<ProvisionAsync>d__6.MoveNext() in C:\workspaces\master\src\github.com\docker\pinata\win\src\Docker.Desktop\Engines\WSL2\WSL2Provisioning.cs:line 44
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Docker.Engines.WSL2.LinuxWSL2Engine.<DoStartAsync>d__23.MoveNext() in C:\workspaces\master\src\github.com\docker\pinata\win\src\Docker.Desktop\Engines\WSL2\LinuxWSL2Engine.cs:line 91
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Docker.ApiServices.StateMachines.TaskExtensions.<WrapAsyncInCancellationException>d__0.MoveNext() in C:\workspaces\master\src\github.com\docker\pinata\win\src\Docker.ApiServices\StateMachines\TaskExtensions.cs:line 29
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Docker.ApiServices.StateMachines.StartTransition.<DoRunAsync>d__5.MoveNext() in C:\workspaces\master\src\github.com\docker\pinata\win\src\Docker.ApiServices\StateMachines\StartTransition.cs:line 67
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at Docker.ApiServices.StateMachines.StartTransition.<DoRunAsync>d__5.MoveNext() in C:\workspaces\master\src\github.com\docker\pinata\win\src\Docker.ApiServices\StateMachines\StartTransition.cs:line 92

**Comment by shermes641** [comment]
I am also on Win 10 2004  and have the same timeout. I disabled WLS2, and still timeout

**Comment by shermes641** [comment]
For me disabling Windows sandbox fixed the issue

**Comment by djs55** [comment]
The fix from edge 2.3.2.0 has now been released in [stable 2.4.0.0](https://desktop.docker.com/win/stable/48506/Docker%20Desktop%20Installer.exe).
